bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxGenericBindingProvider.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.powermax.internal;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.powermax.PowerMaxBindingConfig;
+import org.openhab.binding.powermax.PowerMaxBindingProvider;
+import org.openhab.core.items.Item;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ *
+ * @author lolodomo
+ * @since 1.9.0
+ */
+public class PowerMaxGenericBindingProvider extends AbstractGenericBindingProvider implements PowerMaxBindingProvider {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getBindingType() {
+        return "powermax";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+
+        String selector[] = StringUtils.split(StringUtils.trim(bindingConfig), ':');
+        try {
+            PowerMaxSelectorType.fromString(selector[0]);
+        } catch (IllegalArgumentException e) {
+            throw new BindingConfigParseException("item '" + item.getName() + "': invalid selector " + bindingConfig);
```

Comment:
If the only thing one should do in a `validateItemType` override is validate the item type, then this check should instead happen in `processBindingConfiguration`.